### ISSUE TITLE
Regression in arguments-differ (W0222) from 1.6 to 1.7

### ISSUE THREAD

**Issue by srikanth-viswanathan** [PyCQA/pylint#1531]
### Steps to reproduce
1. Here's a sample program and `pylintrc` that exhibit the behavior: https://gist.github.com/srikanth-viswanathan/d40169518747358b0bef7428f95a2edb
2. With pylint==1.6 installed, `pylint test.py` runs cleanly
3. With pylint==1.7.0 or 1.7.1 installed, `pylint tesy.py` reports `************* Module test
W:  4, 4: Parameters differ from overridden 'data_received' method (arguments-differ)`

### Current behavior
`pylint 1.7+` falsely complains about `W0222`

### Expected behavior
Should not complain when the argument list matches the parent class method

### pylint --version output

Bad version:
```$ pylint --version
pylint 1.7.1, 
astroid 1.5.3
Python 2.7.10 (default, Oct 23 2015, 18:05:06) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
```

Working version:
```$ pylint --version
pylint 1.6.0, 
astroid 1.4.9
Python 2.7.10 (default, Oct 23 2015, 18:05:06) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
```

**Comment by srikanth-viswanathan** [comment]
Any update on this one? Thanks!

**Comment by PCManticore** [comment]
There is no update, no one had time to look over it.

**Comment by PCManticore** [comment]
Renaming `data` to `chunk`, as the original name of the parameter in the parent's implementation, should make the issue go away. We did this to help maintaining correct signatures while working in a multiple dispatching hierarchy, where changing the name of a parameter in a child class could mean that it won't be accessible to be called with the same name of the parameter from the parent's class.
